odoo/addons/test_lint/tests/_odoo_checker_sql_injection.py

```diff
@@ -45,10 +45,27 @@ def _allowable(self, node):
         return (isinstance(node, astroid.Attribute)
             and isinstance(node.expr, astroid.Name)
             and node.attrname.startswith('_')
+            # cr.execute('SELECT * FROM %s' % 'table') is OK since that is a constant
+            or isinstance(node, astroid.Const)
         )
 
     def _check_concatenation(self, node):
         if isinstance(node, astroid.BinOp) and node.op in ('%', '+'):
+            if (isinstance(node.left, astroid.BinOp) and self._check_concatenation(node.left)):
```

Comment:
@xmo-odoo 
I just have detected this case

    cr.execute('SELECT ' + operator + ' FROM table' + 'WHERE')

The representation tree is:

    node.repr_tree()
    BinOp(
    op='+',
    left=BinOp(
        op='+',
        left=BinOp(
            op='+',
            left=Const(value='SELECT '),
            right=Name(name='operator')),
        right=Const(value=' FROM table')),
    right=Const(value='WHERE'))

Notice that left node is another BinOp node
So, it need to be considered recursively

Reply:
Maybe it would make more sense to add that as the final check with an `_allowable` gating? And that would avoid checking `node` against `astroid.Call` which is completely unnecessary as we've already established it's a `astroid.BinOp` e.g. instead of this at line 62/79 have
```python
return (not self._allowable(node.left)) and self._check_concatenation(node.left)
```
or maybe more legibly:
```python
elif self._allowable(node.left):
    return False
return self._check_concatenation(node.left)
```

Reply:
@xmo-odoo 

Sorry, I didn't get you

Are you sure the lines 62/79 are the ones you are talking about?

Even better, do you think you could use this feature, please?
 - <img width="440" alt="Screen Shot 2021-10-01 at 10 45 58" src="https://user-images.githubusercontent.com/6644187/135649507-7554de1d-9b2d-4674-99cc-fb05a03181df.png">

Reply:
I think I got your point

Let me know if it is ok